File: CS-Examples/19_PivotTables/ShowSubTotals.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace ShowSubTotals
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load an Excel file including pivot table
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ShowSubTotals.xlsx");

            //Get the sheet in which the pivot table is located
            Worksheet sheet = workbook.Worksheets["Pivot Table"];

            XlsPivotTable pt = sheet.PivotTables[0] as XlsPivotTable;

            //Show Subtotals
            pt.ShowSubtotals = true;

            String result = "ShowSubTotals_result.xlsx";

            //Save to file
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/19_PivotTables/ShowSubTotals.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace ShowSubTotals
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook
            Dim workbook As New Workbook()

            ' Load an existing Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ShowSubTotals.xlsx")

            ' Get the worksheet named "Pivot Table"
            Dim sheet As Worksheet = workbook.Worksheets("Pivot Table")

            ' Get the first PivotTable from the worksheet and cast it to XlsPivotTable
            Dim pt As XlsPivotTable = TryCast(sheet.PivotTables(0), XlsPivotTable)

            ' Enable showing subtotals in the PivotTable
            pt.ShowSubtotals = True

            ' Specify the name of the resulting file
            Dim result As String = "ShowSubTotals_result.xlsx"

            ' Save the modified workbook to a new file with the specified name and Excel version (2010)
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
